conda_smithy/cli.py

```diff
@@ -214,7 +216,15 @@ def __call__(self, args):
         from conda_smithy import ci_register
 
         owner = args.user or args.organization
-        repo = os.path.basename(os.path.abspath(args.feedstock_directory))
+        meta = conda_build.api.render(
+            args.feedstock_directory,
+            permit_undefined_jinja=True,
+            finalize=False,
+            bypass_env_check=True,
+            trim_skip=False,
+        )[0][0]
```

Comment:
This call is problematic. Some recipes may not render properly without using the exclusive config files in the ci_support dir but ofc those files will not be there. 

I think you need to parse the meta.yaml by hand until the extras section and get the name from there

Reply:
Can the following be used instead (as is done in the `Init.__call__` method):

```python
meta = MetaData(args.feedstock_directory)
```

Reply:
I’d checkout the code we use to get the team from the meta yaml in the web services bot. I think this will be more robust. Or you could use a regex.

Reply:
This is the same code as in https://github.com/conda-forge/conda-smithy/blob/19f019443139e34c530cd07f5e7939040387a986/conda_smithy/github.py#L107-L115

If it would fail, we would not even be able to generate a github repository and would not get to this step.

Reply:
OK. I'd say it is never a good idea to parse recipes like that if they have not been fully rendered properly. A regex will be much more robust in both places.

Reply:
A regex is never robust when the language is not a regular language.

Reply:
I mean in this case, we can read the file until the extra section and then look, but I am not going to push this.